src/mainwindow.cpp

```diff
@@ -201,6 +201,12 @@ void MainWindow::run()
 
 void MainWindow::run(int index)
 {
+    if (index < 0 || index >= testcases->count())
+    {
+        LOG_WARN(INFO_OF(index) << INFO_OF(testcases->count()));
+        return;
+    }
+
```

Comment:
When can indices be out of range?

Reply:
When we make mistakes. It shouldn't be out of range.

Reply:
Okay, so it's for finding issues quickly.